modules/openxr/scene/openxr_render_model_manager.h

```diff
@@ -0,0 +1,84 @@
+/**************************************************************************/
+/* Copyright (c) 2014-present Godot Engine contributors (see AUTHORS.md). */
+/* Copyright (c) 2007-2014 Juan Linietsky, Ariel Manzur.                  */
+/*                                                                        */
+/* Permission is hereby granted, free of charge, to any person obtaining  */
+/* a copy of this software and associated documentation files (the        */
+/* "Software"), to deal in the Software without restriction, including    */
+/* without limitation the rights to use, copy, modify, merge, publish,    */
+/* distribute, sublicense, and/or sell copies of the Software, and to     */
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#pragma once
+
+#include "openxr_render_model.h"
+
+#include "scene/3d/node_3d.h"
+#include "scene/resources/packed_scene.h"
+#include "servers/xr/xr_positional_tracker.h"
+
+#include <openxr/openxr.h>
+
+class OpenXRRenderModelManager : public Node3D {
+	GDCLASS(OpenXRRenderModelManager, Node3D);
+
+public:
+	enum RenderModelTracker {
+		RENDER_MODEL_TRACKER_ANY,
+		RENDER_MODEL_TRACKER_NONE_SET,
+		RENDER_MODEL_TRACKER_LEFT_HAND,
+		RENDER_MODEL_TRACKER_RIGHT_HAND,
+	};
+
+	virtual PackedStringArray get_configuration_warnings() const override;
+
+	void set_tracker(RenderModelTracker p_tracker);
+	RenderModelTracker get_tracker() const;
+
+	void set_make_local_to_pose(const String &p_action);
+	String get_make_local_to_pose() const;
+
+private:
+	HashMap<RID, Node3D *> render_models;
+	Node3D *container = nullptr;
+
+	bool is_dirty = false;
+	RenderModelTracker tracker = RENDER_MODEL_TRACKER_ANY;
+	String make_local_to_pose;
+
+	// cached values
+	Ref<XRPositionalTracker> positional_tracker;
+	XrPath xr_path = XR_NULL_PATH;
+
+	bool _has_filters();
+	void _on_render_model_added(OpenXRRenderModel *p_render_model);
+	void _on_render_model_removed(OpenXRRenderModel *p_render_model);
```

Comment:
I think these are supposed to take `RID`s rather than objects:

```suggestion
	void _on_render_model_added(RID p_render_model);
	void _on_render_model_removed(RID p_render_model);
```

It's a little confusing because `RenderModelManager` has its own signals with the same names that will get a `OpenXRRenderModel *` obect, but these are used to connect to the signals on `OpenXRRenderModelExtension`. which pass an `RID`

Reply:
GRMBL, I was looking at the wrong signal!! `OpenXRRenderModelManager` also emits a signal with the same name.
I'm so dumb!

Reply:
Looks good to me now!